### PR TITLE
NFS over REST

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 1]
```diff
@@ -3403,6 +3415,35 @@
...
                 "description": "The parent fileId of the file."
+              },
+              "x-ms-mode": {
+                "x-ms-client-name": "fileMode",
+                "type": "string",
+                "description": "NFS only. The mode of the file or directory."
+              },
+              "x-ms-owner": {
+                "x-ms-client-name": "owner",
+                "type": "string",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:3426]: This type is wrong, fix to match the owner parameter

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 2]
```diff
@@ -3403,6 +3415,35 @@
...
+                "description": "NFS only. The owner of the file or directory."
+              },
+              "x-ms-group": {
+                "x-ms-client-name": "group",
+                "type": "string",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:3431]: This type is wrong, fix to match the group parameter

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 3]
```diff
@@ -3403,6 +3415,39 @@
...
                 "description": "The parent fileId of the file."
+              },
+              "x-ms-mode": {
+                "x-ms-client-name": "fileMode",
+                "type": "string",
+                "description": "NFS only. The mode of the file or directory."
+              },
+              "x-ms-owner": {
+                "x-ms-client-name": "owner",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:3426]: Change to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 4]
```diff
@@ -3403,6 +3415,39 @@
...
+                "minimum": 0,
+                "format": "int64",
+                "description": "NFS only. The owner of the file or directory."
+              },
+              "x-ms-group": {
+                "x-ms-client-name": "group",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:3433]: Change to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 5]
```diff
@@ -7880,6 +7925,55 @@
       "in": "header",
       "required": false,
       "type": "boolean"
+    },
+    "Owner": {
+      "name": "x-ms-owner",
+      "x-ms-client-name": "owner",
+      "in": "header",
+      "required": false,
+      "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:7934]: Change to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 6]
```diff
@@ -7880,6 +7925,55 @@
...
+      "minimum": 0,
+      "format": "int64",
+      "description": "Optional, NFS only. The owner of the file or directory.",
+      "x-ms-parameter-location": "method"
+    },
+    "Group": {
+      "name": "x-ms-group",
+      "x-ms-client-name": "group",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:7942]: Change to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 7]
```diff
@@ -3663,6 +3708,31 @@
...
+              "x-ms-owner": {
+                "x-ms-client-name": "owner",
+                "type": "integer",
+                "minimum": 0,
+                "format": "int64",
+                "description": "NFS only. The owner of the file or directory."
+              },
+              "x-ms-group": {
+                "x-ms-client-name": "group",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:3726]: Considering changing to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 8]
```diff
@@ -3663,6 +3708,31 @@
...
                 "description": "The length of the blob/file content inside the message body when the response body is returned as a structured message. Will always be smaller than Content-Length."
+              },
+              "x-ms-mode": {
+                "x-ms-client-name": "fileMode",
+                "type": "string",
+                "description": "NFS only. The mode of the file or directory."
+              },
+              "x-ms-owner": {
+                "x-ms-client-name": "owner",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:3719]: Considering changing to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 9]
```diff
@@ -3883,6 +3953,31 @@
...
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:3964]: Considering changing to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 10]
```diff
@@ -3883,6 +3953,31 @@
...
+                "minimum": 0,
+                "format": "int64",
+                "description": "NFS only. The owner of the file or directory."
+              },
+              "x-ms-group": {
+                "x-ms-client-name": "group",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:3971]: Considering changing to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 11]
```diff
@@ -4124,6 +4219,45 @@
...
                 }
+              },
+              "x-ms-mode": {
+                "x-ms-client-name": "fileMode",
+                "type": "string",
+                "description": "NFS only. The mode of the file or directory."
+              },
+              "x-ms-owner": {
+                "x-ms-client-name": "owner",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:4230]: Considering changing to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 12]
```diff
@@ -4124,6 +4219,45 @@
...
+                "minimum": 0,
+                "format": "int64",
+                "description": "NFS only. The owner of the file or directory."
+              },
+              "x-ms-group": {
+                "x-ms-client-name": "group",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:4237]: Considering changing to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 13]
```diff
@@ -4353,6 +4502,31 @@
...
                 "description": "The parent fileId of the directory."
+              },
+              "x-ms-mode": {
+                "x-ms-client-name": "fileMode",
+                "type": "string",
+                "description": "NFS only. The mode of the file or directory."
+              },
+              "x-ms-owner": {
+                "x-ms-client-name": "owner",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:4513]: Considering changing to string to pattern match Data Lake

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json [hunk 14]
```diff
@@ -4353,6 +4502,31 @@
...
+                "minimum": 0,
+                "format": "int64",
+                "description": "NFS only. The owner of the file or directory."
+              },
+              "x-ms-group": {
+                "x-ms-client-name": "group",
+                "type": "integer",
```
Comment (seanmcc-msft) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json:4520]: Considering changing to string to pattern match Data Lake

**Review by weirongw23-msft — APPROVED** [review]

**Review by jaschrep-msft — APPROVED** [review]

**Pull request merged**